pallets/ddc-nodes/src/tests.rs

```diff
@@ -60,7 +96,162 @@ fn create_node_works() {
+			NodeParams::CDNParams(cdn_node_params)
+		));
+
+		// Check storage
+		assert!(CDNNodes::<Test>::contains_key(node_pub_key.clone()));
+		assert!(DdcNodes::exists(&NodePubKey::CDNPubKey(node_pub_key.clone())));
+		if let Ok(cluster_id) =
+			DdcNodes::get_cluster_id(&NodePubKey::CDNPubKey(node_pub_key.clone()))
+		{
+			assert_eq!(cluster_id, None);
+		}
+		let cdn_node = DdcNodes::cdn_nodes(&node_pub_key).unwrap();
+		assert_eq!(cdn_node.pub_key, node_pub_key);
+	})
+}
+
+#[test]
+fn create_storage_node_works() {
+	ExtBuilder.build_and_execute(|| {
+		System::set_block_number(1);
+		let bytes = [0u8; 32];
+		let node_pub_key = AccountId32::from(bytes);
+		let storage_node_params = StorageNodeParams {
+			host: vec![1u8, 255],
+			http_port: 35000u16,
+			grpc_port: 25000u16,
+			p2p_port: 15000u16,
+		};
+
+		// Node params are not valid
+		assert_noop!(
+			DdcNodes::create_node(
+				RuntimeOrigin::signed(1),
+				NodePubKey::StoragePubKey(node_pub_key.clone()),
+				NodeParams::CDNParams(CDNNodeParams {
+					host: vec![1u8, 255],
+					http_port: 35000u16,
+					grpc_port: 25000u16,
+					p2p_port: 15000u16,
+				})
+			),
+			Error::<Test>::InvalidNodeParams
+		);
+
+		// Pub key invalid
+		assert_noop!(
+			StorageNode::<Test>::new(
+				NodePubKey::CDNPubKey(node_pub_key.clone()),
+				1u64,
+				NodeParams::StorageParams(storage_node_params.clone())
+			),
+			NodeError::InvalidStorageNodePubKey
+		);
+
+		// Host length exceeds limit
+		assert_noop!(
+			DdcNodes::create_node(
+				RuntimeOrigin::signed(1),
+				NodePubKey::StoragePubKey(node_pub_key.clone()),
+				NodeParams::StorageParams(StorageNodeParams {
+					host: vec![1u8; 256],
+					http_port: 35000u16,
+					grpc_port: 25000u16,
+					p2p_port: 15000u16,
+				})
+			),
+			Error::<Test>::HostLenExceedsLimit
+		);
+
+		// Node created
+		assert_ok!(DdcNodes::create_node(
+			RuntimeOrigin::signed(1),
+			NodePubKey::StoragePubKey(node_pub_key.clone()),
+			NodeParams::StorageParams(storage_node_params.clone())
+		));
+
+		// Check storage
+		assert!(StorageNodes::<Test>::contains_key(node_pub_key.clone()));
+		assert!(DdcNodes::exists(&NodePubKey::StoragePubKey(node_pub_key.clone())));
+		if let Ok(cluster_id) =
+			DdcNodes::get_cluster_id(&NodePubKey::StoragePubKey(node_pub_key.clone()))
+		{
+			assert_eq!(cluster_id, None);
+		}
+		let storage_node = DdcNodes::storage_nodes(&node_pub_key).unwrap();
+		assert_eq!(storage_node.pub_key, node_pub_key);
+
+		// Node already exists
+		assert_noop!(
+			DdcNodes::create_node(
+				RuntimeOrigin::signed(1),
+				NodePubKey::StoragePubKey(node_pub_key.clone()),
+				NodeParams::StorageParams(storage_node_params)
+			),
+			Error::<Test>::NodeAlreadyExists
+		);
+
+		// Checking that event was emitted
+		assert_eq!(System::events().len(), 1);
+		System::assert_last_event(
+			Event::NodeCreated { node_pub_key: NodePubKey::StoragePubKey(node_pub_key) }.into(),
+		)
+	})
+}
+
+#[test]
+fn create_storage_node_with_node_creator() {
```

Comment:
I think it is better to just use the `NodeCreator` trait in the `create_node` extrinsic's body as `<Self as NodeCreator<T>>::create_node(node_pub_key, provider_id, node_params)` and this way avoid code duplication and tests duplication for the same node creation scenario

Reply:
agree, tests need to be more clean and reusable